csharp/extractor/Semmle.Extraction.CSharp/Entities/Method.AstLineCounter.cs

```diff
@@ -4,9 +4,9 @@
 using Semmle.Util;
 using System.IO;
 
-namespace Semmle.Extraction.CSharp.Populators
+namespace Semmle.Extraction.CSharp.Entities
 {
-    public static class MethodExtensions
+    public abstract partial class Method
```

Comment:
I would prefer extension methods over `partial` classes.

Reply:
I could make `AstLineCounter` top level and public, and then there's no partial, and no extension method either. What do you think? I'm not convinced about all the extension methods that we have. They make sense when they extend Roslyn classes, but why have extension methods that extend classes that we directly control?

Also, in this particular case, `NumberOfLines` was an extension on `Context`, and it didn't use `Context` at all.

Reply:
Then I prefer making `AstLineCounter` public.